src/main/java/graphql/execution/ExecutionContext.java

```diff
@@ -35,13 +36,14 @@
     private final Object context;
     private final Instrumentation instrumentation;
     private final List<GraphQLError> errors = new CopyOnWriteArrayList<>();
+    private final DataLoaderRegistry dataLoaderRegistry;
     private final DeferSupport deferSupport = new DeferSupport();
 
     public ExecutionContext(Instrumentation instrumentation, ExecutionId executionId, GraphQLSchema graphQLSchema, InstrumentationState instrumentationState, ExecutionStrategy queryStrategy, ExecutionStrategy mutationStrategy, ExecutionStrategy subscriptionStrategy, Map<String, FragmentDefinition> fragmentsByName, Document document, OperationDefinition operationDefinition, Map<String, Object> variables, Object context, Object root) {
```

Comment:
can we mark the constructor as Internal and I am not sure we should always create a new DLR here.

Reply:
sure

Reply:
I did the extra new DLR here JUST in case some one created it by hand because it was a public method

Reply:
But since this is already a breaking change - I am going to clean this up by removing the old public constructor.

Our code should be the only constructor of these things